crates/cext/src/circuit.rs

```diff
@@ -68,6 +69,186 @@ pub extern "C" fn qk_circuit_new(num_qubits: u32, num_clbits: u32) -> *mut Circu
     Box::into_raw(Box::new(circuit))
 }
 
+/// @ingroup QkCircuit
```

Comment:
Speaking of documentation, does it make sense to have all register-related methods be part of the `QkCircuitGroup`? Are we representing `Qk{Classical, Quantum}Register` as a datatype for `QkCircuit` or should it have its own documentation section? I'd lean more towards the latter option since they are pointing to different structures, which should have their own documentation.

Reply:
Given that the circuits page probably becomes large, I agree a different one might be nicer. We initially thought of having one struct per docs page, like some other larger C projects.

Reply:
I split out the docs in: https://github.com/Qiskit/qiskit/pull/14355/commits/0d0e474accea28a02a00f88120b43bb14e9ceaee